### PR TITLE
Inference subclass

### PR DESCRIPTION
**Pull request by CorentinSeznec**
Inférence avec poesy. Création d'un InferPoesyDataset qui hérite de PoesyDataset.  Normalement, le code interfère le moins possible avec le code d'origine

### CODE REVIEW

**Review by flyIchtus — DISMISSED** [review]

File: py4cast/datasets/poesy.py [hunk 1]
```diff
@@ -811,6 +830,114 @@ def prepare(cls, path_config: Path):
         return train_ds
 
 
+class InferPoesyDataset(PoesyDataset):
```
[the remark this answers is not in the document]
Comment (CorentinSeznec) [reply py4cast/datasets/poesy.py:837]: Done.

File: bin/inference.py [hunk 2]
```diff
@@ -26,8 +26,8 @@
     hparams = lightning_module.hparams["hparams"]
 
     # Get dataset for inference
-    _, _, test_ds = get_datasets(
-        hparams.dataset_name,
+    infer_ds, _, _ = get_datasets(
+        "poesy_infer",
```
Comment (colon3ltocard) [inline bin/inference.py:30]: Be careful here you have hardcoded the dataset name, this means we cannot do titan inference anymore. Also you changed the index of inference dataset (test in our case). Could you fix that ?

Comment (CorentinSeznec) [reply bin/inference.py:30]: Ok for the index, thanks for noticing it. But I have a question about the dataset name. Currently, I have modified only the dataset Poesy (that is why it is hardcoded). If we want to preserve the proper functioning of the inference on Titan, I should modifiy it as well.
Is it okay for you if I try to modify it the same way I modify Poesy ? It means to create TitanInferDataset and InferSample. And also to dissociate num_pred_steps variable in TitanSettings as num_output_steps and num_inference_pred_steps.

Comment (colon3ltocard) [reply bin/inference.py:30]: @CorentinSeznec let's not modify TITAN in this PR I would like to ask @LBerth first. If you could fix the index and restore the name arg passing in this PR I can merge it and we see next week what we do about TITAN. Thanks !

Comment (CorentinSeznec) [reply bin/inference.py]: @colon3ltocard Ok, it is done. Thank you !

File: py4cast/datasets/__init__.py [hunk 3]
```diff
@@ -33,13 +33,33 @@
...
+try:
+    from .titan import InferTitanDataset
+
+    registry["titan_infer"] = (
+        InferTitanDataset,
+        default_config_root / "titan_infer.json",
+    )
+except ImportError:
+    warnings.warn(f"Could not import InferTitanDataset. {traceback.format_exc()}")
+
```
Comment (colon3ltocard) [inline py4cast/datasets/__init__.py:45]: I think you didn't want to commit this one ?

Comment (CorentinSeznec) [reply py4cast/datasets/__init__.py]: No, indeed, it will be for the titan inference. I fixed it. Thank you.

**Review by colon3ltocard — APPROVED** [review]

File: config/datasets/poesy.json [hunk 4]
```diff
@@ -61,5 +61,6 @@
                 "WIND.V.PHYS":{"shortname":"v","level": [1]}
                 }
             }
-        }
+        },
+    "num_inference_pred_steps": 3
```
Comment (A669015) [inline config/datasets/poesy.json:65]: Is this now madatory to add this parameter in all Poesy config file ? There is no way to make it optional, to not change this file for example ?

Comment (CorentinSeznec) [reply config/datasets/poesy.json:65]: Yes, you're right, I will take it off. Thank you !

**Review by A669015 — APPROVED** [review]

**Comment by flyIchtus** [comment]
Cannot merge, wait for corrections pls

**Comment by colon3ltocard** [comment]
@CorentinSeznec @flyIchtus coul you document your inference script in the README.md ?

**Review by colon3ltocard — APPROVED** [review]

**Review by flyIchtus — APPROVED** [review]

**Pull request merged**